src/clang.rs

```diff
@@ -276,6 +276,58 @@ impl Cursor {
         true
     }
 
+    /// Is the referent any kind of template parameter?
+    pub fn is_template_parameter(&self) -> bool {
+        match self.kind() {
+            CXCursor_TemplateTemplateParameter |
+            CXCursor_TemplateTypeParameter |
+            CXCursor_NonTypeTemplateParameter => true,
+            _ => false,
+        }
+    }
+
+    /// Does the referent's type or value depend on a template parameter?
+    pub fn is_dependent_on_template_parameter(&self) -> bool {
```

Comment:
I wonder if it'd be cleaner to just merge this on `bit_width` itself (or make it private).

Basically doing something like:

```rust
fn bit_width(&self) -> Option<u32> {
    let expr = self.bit_width_expr()?;
    if expr.is_dependent_on_template_parameter() {
        return None;
    }
    // ...
}
```

It seems that'd simplify consumers (basically, if it's a bitfield, but can't get the width, then it's template-dependnt).

Reply:
Good point, this simplified it a lot.

I also renamed the `CompInfo` field to `has_unevaluable_bit_field_width` since now it doesn't know/care why specifically it couldn't be evaluated.